app/src/main/java/com/github/meeplemeet/ui/space_renter/CreateSpaceRenterScreen.kt

```
@@ -0,0 +1,336 @@
+        derivedStateOf {
+          name.isNotBlank() &&
+              isValidEmail(email) &&
+              addressText.isNotBlank() &&
+              hasOpeningHours &&
+              hasAtLeastOneSpace &&
+              allSpacesValid
+        }
+      }
+
+  // Keep address text in sync with the location search bar
+  LaunchedEffect(locationUi.locationQuery) {
+    if (locationUi.locationQuery.isNotEmpty() && addressText != locationUi.locationQuery) {
+      addressText = locationUi.locationQuery
+    }
+  }
+
+  fun discardAndBack() {
+    name = ""
+    email = ""
+    phone = ""
+    link = ""
+    addressText = ""
+    selectedLocation = null
+    week = emptyWeek()
+    editingDay = null
+    showHoursDialog = false
+    spaces = emptyList()
+    onBack()
+  }
+
+  fun addSpace() {
+    spaces = spaces + Space(seats = 1, costPerHour = 0.0)
+  }
+
+  Scaffold(
+      topBar = {
+        CenterAlignedTopAppBar(
+            title = {
+              Text(
+                  AddSpaceRenterUi.Strings.SCREEN_TITLE,
+                  modifier = Modifier.testTag(CreateSpaceRenterScreenTestTags.TITLE))
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = onBack,
+                  modifier = Modifier.testTag(CreateSpaceRenterScreenTestTags.NAV_BACK)) {
+                    Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back")
+                  }
+            },
+            modifier = Modifier.testTag(CreateSpaceRenterScreenTestTags.TOPBAR))
+      },
+      snackbarHost = {
+        SnackbarHost(
+            snackbarHost,
+            modifier = Modifier.testTag(CreateSpaceRenterScreenTestTags.SNACKBAR_HOST))
+      },
+      bottomBar = {
+        ActionBar(
+            onDiscard = { discardAndBack() },
+            onPrimary = {
+              val addr = selectedLocation ?: Location(name = addressText)
```

I don't think it's smart to have a fallback to a hardcoded `Location()` with the inputted text from the user. If something is optional in the ViewModel then don't give it a fallback, and if it's required then the form cannot be validated until all required fields are filled.

---

The problem is that either the viewmodel and the spaceRenter.kt use non-nullable location. But you are right, the code didn't checked well enough if a selected location was setup or not.

The problem was solved in commit https://github.com/Meeple-Meet/Meeple-Meet/pull/138/commits/9b5ead2132d1b0db8bdcb5e3e8608e6477eb708d